### DISCUSSION TITLE
How to write custom exception messages  if an element is not available in selenide

### DISCUSSION THREAD

**Discussion by aravindraveendran7** [selenide/selenide#2466] [General]
Hello

I need to wait for a web element, say 5 seconds and if it is present then click, or else need to throw a custom exception which I define.I have tried with element. exists() which doesn't wait for the element in result, if the element is present it doesn't click on the same. For element. is(condition. enabled) it throws an exception if an element is not available.

In a nutshell,I don't want the exception like element not found for the xpath....... rather I want something like an element not found due to so and so reason as a custom exception which I can define. How can this be achieved? please advise.

This is the piece of code I have tried, here element contains the xpath to be clicked.This works only if the element is present.If the element is not displayed it throws no such element exception which I would like to avoid.Any help appreciated.

```java
if (element.shouldBe(enabled,Duration.ofSeconds(5)).exists()) {
    element.shouldBe(visible, enabled).scrollIntoView("{ behavior: 'instant', block: 'end', inline: 'nearest' }").click(usingJavaScript());
} else {
    isenabled = false;
}
```

**Comment by asolntsev** [comment]
@aravindraveendran7 Hi. It's a typical question "how to close a banner if it can randomly appear any moment?"

Here is my answer:
https://selenide.org/2019/12/02/advent-calendar-how-to-abuse-selenide/
